timelock-agent/src/main/java/com/palantir/timelock/paxos/LeaderPingHealthCheck.java

```diff
@@ -51,7 +50,7 @@ public LeaderPingHealthCheck(
         this.pingers = pingers;
         this.executorService = PTExecutors.newFixedThreadPool(
                 pingers.size(),
-                new NamedThreadFactory("leader-ping-healthcheck", true));
+                "leader-ping-healthcheck");
```

Comment:
This one seems a bit suspicious - we're changing it to be non-daemon, and there doesn't seem to be anything that shuts down the executor service anywhere. Do we know/have we tested if timelock will shut down gracefully?

Reply:
Ah, I had though the default NamedThreadFactory ctor used daemon threads, but that was incorrect. I definitely want this to use daemon threads.

I don't think we actually want any ptexecutors to use non-daemon threads since they've historically been a source of pain.